tests/unit/modules/network/junos/test_junos_routing_instances.py

```diff
@@ -0,0 +1,403 @@
+
+        self.mock_execute_show_command = patch(
+            "ansible_collections.junipernetworks.junos.plugins.module_utils.network.junos.facts.routing_instances.routing_instances."
+            "Routing_instancesFacts.get_device_data"
+        )
+        self.execute_show_command = self.mock_execute_show_command.start()
+
+    def tearDown(self):
+        super(TestJunosRouting_instancesModule, self).tearDown()
+        self.mock_load_config.stop()
+        self.mock_lock_configuration.stop()
+        self.mock_unlock_configuration.stop()
+        self.mock_commit_configuration.stop()
+        self.mock_execute_show_command.stop()
+
+    def load_fixtures(
+        self, commands=None, format="text", changed=False, filename=None
+    ):
+        def load_from_file(*args, **kwargs):
+            if filename:
+                output = load_fixture(filename)
+            else:
+                output = load_fixture("junos_routing_instances_config.cfg")
+            return output
+
+        self.execute_show_command.side_effect = load_from_file
+
+    def sort_routing_instances(self, entry_list):
+        for entry in entry_list:
+            if entry.get("instance"):
+                entry["routing_instances"].sort(key=lambda i: i.get("name"))
+
+    def test_junos_routing_instances_merged(self):
+        set_module_args(
+            dict(
+                config=[
+                    dict(
+                        name="test",
+                        type="vrf",
+                        route_distinguisher="10.58.255.1:37",
+                        vrf_imports=["test-policy"],
+                        vrf_exports=["test-policy", "test-policy-1"],
+                        interfaces=[
+                            dict(name="sp-0/0/0.0"),
+                            dict(name="gr-0/0/0.0"),
+                        ],
+                        connector_id_advertise=True,
+                    )
+                ],
+                state="merged",
+            )
+        )
+        commands = [
+            '<nc:routing-instances xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0">'
+            "<nc:instance><nc:name>test</nc:name><nc:connector-id-advertise/>"
+            "<nc:instance-type>vrf</nc:instance-type><nc:interface><nc:name>sp-0/0/0.0</nc:name>"
+            "</nc:interface><nc:interface><nc:name>gr-0/0/0.0</nc:name>"
+            "</nc:interface><nc:route-distinguisher><nc:rd-type>10.58.255.1:37</nc:rd-type>"
+            "</nc:route-distinguisher><nc:vrf-import>test-policy</nc:vrf-import>"
+            "<nc:vrf-export>test-policy</nc:vrf-export><nc:vrf-export>test-policy-1</nc:vrf-export>"
+            "</nc:instance></nc:routing-instances>"
+        ]
+        result = self.execute_module(changed=True, commands=commands)
+        self.assertEqual(sorted(result["commands"]), sorted(commands))
+
+    def test_junos_routing_instances_merged_idempotent(self):
+        set_module_args(
+            dict(
+                config=[
```

Comment:
@rohitthakur2590 have u covered all of the module params accepted by the module?

Reply:
big xml trees are generated, if required I'll be updating that in different PR

Reply:
yes , its covered between units and integration tests